mill

```diff
@@ -48,4 +48,4 @@ fi
 unset MILL_DOWNLOAD_PATH
 unset MILL_VERSION
 
-exec $MILL_EXEC_PATH "$@"
+exec $MILL_EXEC_PATH -D "mill.main.cli=${0}" "$@"
```

Comment:
Thanks for working on this @lefou! So quick question just to make sure I understand. In the situation where a Metals user just lets Metals generate the BSP config  (and let's say there is no millw in the repo), then Metals will use the millw it has in it's resources to do so. When we use the updated one, what is the actual value of `mill.main.cli` here then? I'm not sure I fully understand.

Reply:
Good question. I also have a PR to [`millw`](https://github.com/lefou/millw) which is used by Metals. So, it's rather a question what the actual value of the shell variable `${0}` will look like, which depends on how `millw` it is handled by Metals. It may be a problem, if the script is invoked from an temporary (non-permatent) location.

* https://github.com/lefou/millw/pull/41

Reply:
Let me use that pr from `millw` in Metals and just try it out. I'll report back.

Reply:
Just as an idea, if there is need for a change: Metals could place the `millw` script inside `.metals` so it stays permanent for subsequent runs.

Reply:
So this ended up being way harder to test than I thought due to how we do stuff in Metals. All that to say, we will however end up using `millw` inside of a temp directory that we create. I'm looking into now if we can instead just do as you mention and copy it into `.metals/millw` and just use it there.